StRoot/StEvent/StFcsHit.cxx

```diff
@@ -73,7 +73,7 @@ int   StFcsHit::nPeak()    const {return mNPeak;}
 float StFcsHit::energy()   const {return mEnergy;}
 
 void StFcsHit::setDepCh(unsigned short ns, unsigned short ehp, unsigned short dep, unsigned short ch) { 
-    mDepCh = (ns << 15) | (ehp<<13) | (dep<<8) | ch; 
+    mDepCh = ((ns&0x1) << 15) | ((ehp&0x3)<<13) | ((dep&0x1f)<<8) | (ch & 0xff);
```

Comment:
```suggestion
    assert((ns & (~0x1)) == 0);
    assert((ehp & (~0x3)) == 0);
    assert((dep & (~0x1f)) == 0);
    assert((ch & (~0xff)) == 0);
    mDepCh = ((ns&0x1) << 15) | ((ehp&0x3)<<13) | ((dep&0x1f)<<8) | (ch & 0xff);
```

Reply:
I thought using assert was strongly discouraged...

Reply:
Why?

Reply:
Stpping the whole chain and delaying QA process for all other detectors is worse than one detector getting a bugged data?

Reply:
Turns out we don't define the `NDEBUG` for release builds, so this was not a good suggestion.

Crashing the chain is better than redoing the production.